### PR TITLE
[Candidate Parameters] make select statement use join instead of subselect

### PR DESCRIPTION
**Pull request by stellarxo**
Change subselect to join

### CODE REVIEW

File: modules/candidate_parameters/ajax/getData.php [hunk 1]
```diff
@@ -189,9 +189,9 @@ function getFamilyInfoFields()
     );
 
     $siblingsList = $db->pselect(
-        "SELECT CandID 
-        FROM family 
-        WHERE FamilyID=(SELECT FamilyID FROM family WHERE CandID=:candid)",
+        "SELECT f1.CandID 
+        FROM family f1 JOIN family f2
+        WHERE f1.FamilyID=f2.FamilyID and f2.CandId=:candid GROUP BY f1.CandID;",
```
Comment (driusan) [inline modules/candidate_parameters/ajax/getData.php]: I think this should be a JOIN condition instead of a where clause (just the FamilyID part)

Comment (stellarxo) [inline modules/candidate_parameters/ajax/getData.php]: Sorry! Do you mean how I did it now?

**Comment by xlecours** [comment]
This allow a Candidate to be part of multiple families and will return all siblings from those families without error.

**Comment by stellarxo** [comment]
@xlecours that was the previous functionality. I'm not sure how it should be changed

**Comment by xlecours** [comment]
This script allow can used to access candidates info regardless of permission.
RM#11614

File: modules/candidate_parameters/ajax/getData.php [hunk 2]
```diff
@@ -215,12 +215,9 @@ function getFamilyInfoFields()
     }
 
     $familyMembers = $db->pselect(
-        "SELECT CandID as FamilyCandID, Relationship_type 
-        FROM family 
-        WHERE FamilyID=(
-          SELECT FamilyID 
-          FROM family 
-          WHERE CandID = :candid) AND CandID <> :candid2 
+        "SELECT f1.CandID as FamilyCandID, Relationship_type
```
Comment (xlecours) [inline modules/candidate_parameters/ajax/getData.php]: This query cause an error: Column 'Relationship_type' in field list is ambiguous
It should be f1.Relationship_type

**Comment by stellarxo** [comment]
@xlecours fixed

File: modules/candidate_parameters/ajax/getData.php [hunk 3]
```diff
@@ -215,12 +215,9 @@ function getFamilyInfoFields()
...
-        "SELECT CandID as FamilyCandID, Relationship_type 
-        FROM family 
-        WHERE FamilyID=(
-          SELECT FamilyID 
-          FROM family 
-          WHERE CandID = :candid) AND CandID <> :candid2 
+        "SELECT f1.CandID as FamilyCandID, f1.Relationship_type 
+        FROM family f1 JOIN family f2 ON f1.FamilyID=f2.FamilyID
+        WHERE f2.CandID = :candid AND f1.CandID <> :candid2 
           ORDER BY CandID",
```
Comment (xlecours) [inline modules/candidate_parameters/ajax/getData.php]: Column 'CandID' in order clause is ambiguous

Comment (xlecours) [inline modules/candidate_parameters/ajax/getData.php]: It should be fine after this last change

**Comment by stellarxo** [comment]
@xlecours fixed

**Comment by stellarxo** [comment]
@driusan there's a 1-to-1 correspondence on the family tables so left join doesn't seem to be needed

**Pull request merged**